hack/choco/update-choco-package.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+ 
+# Copyright 2022 VMware Tanzu Community Edition contributors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+ 
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+ 
+version="${1:?TCE version argument empty. Example usage: ./hack/choco/update-choco-package.sh v0.10.0}"
+: "${GITHUB_TOKEN:?GITHUB_TOKEN is not set}"
+ 
+ 
+temp_dir=$(mktemp -d)
+ 
+pushd "${temp_dir}"
+
+TCE_REPO="https://github.com/vmware-tanzu/community-edition/" 
+TCE_REPO_RELEASES_URL="https://github.com/vmware-tanzu/community-edition/releases"
+TCE_WINDOWS_TAR_BALL_FILE="tce-darwin-amd64-${version}.tar.gz"
+TCE_CHECKSUMS_FILE="tce-checksums.txt"
+ 
+echo "Checking if the necessary files exist for the TCE ${version} release"
+ 
+curl -f -I -L \
+   "${TCE_REPO_RELEASES_URL}/download/${version}/${TCE_WINDOWS_TAR_BALL_FILE}" > /dev/null || {
+       echo "${TCE_WINDOWS_TAR_BALL_FILE} is not accessible in TCE ${version} release"
+       exit 1
+   }
+ 
+wget "${TCE_REPO_RELEASES_URL}/download/${version}/${TCE_CHECKSUMS_FILE}" || {
+   echo "${TCE_CHECKSUMS_FILE} is not accessible in TCE ${version} release"
+   exit 1
+}
+ 
+git clone "${TCE_REPO}"
```

Comment:
Let's do `git clone --depth 1` since we don't need to pull down the full history.

Reply:
Do we need to do a git clone since we're using the `checkout` action?

Reply:
Hey @nrb thanks for reviewing this PR.
I tried this thing but it was giving error that no main branch is there will give an another try and update with logs here.

Reply:
If we use the code that we are already in, then we can just have the `checkout` action and remove the creation of a tmp directory to make this change in.

On the other hand, if we ever want to run this locally, then creating a new pristine temporary clone would help make sure there is nothing else dirty in the repo that would get picked up and committed below.

Reply:
Let's go with pristine directory over the existing git repo from the `checkout` action?

Reply:
Ok, I'm fine with the fresh clone.

Reply:
There is a bug that with `--depth 1` gh pr create will  fail.
Issue comment link for complete context - https://github.com/cli/cli/issues/2979#issuecomment-780490392
So for now we have to avoid   `--depth 1`. 
What do you say @stmcginnis ?

Reply:
That's unfortunate, but at least for now the repo is still fairly new enough that pulling the entire history down isn't that much extra overhead. Good thing to fix in the future if the referenced issue is resolved.

If you are still making updates, can you add a comment above this line stating something to the effect that we want to use `--depth 1` but need to wait until a fix for that issue is available?

Reply:
Sure I will add a comment above this line and reference the issue too.